### DISCUSSION TITLE
Render fisheye?

### DISCUSSION THREAD

**Discussion by StarsTesla** [mitsuba-renderer/mitsuba3#1325] [Feature requests] [answered]
Will mitsuba support fisheye or some distort rendering in the furture?

**Comment by njroussel** [comment, marked as answer]
Hi @StarsTesla 

There is no existing plan to add this feature at the moment.


Howerver, I don't think it would be particularly difficult to add. In fact, I think this would be something that could easily fit into a custom Python `Sensor` plugin in a few lines of code. If anyone is interested in doing this, have a look at https://github.com/rgl-epfl/differentiable-sdf-rendering/blob/main/figures/spherical_sensor.py where a spherical sensor is defined and registered with Mitsuba.
